### PR TITLE
update monitor gui.

### PR DESCRIPTION
**Pull request by Alice1319**
related eclipse-xpanse/xpanse#373
fixed eclipse-xpanse/xpanse#367

image

### CODE REVIEW

File: src/components/content/monitor/MonitorChart.tsx [hunk 1]
```diff
@@ -3,129 +3,184 @@
...
+                            );
+                        })}
+                    </Radio.Group>
+                </div>
+                <Divider type='vertical' className={'charts-divider-class'} />
+                <div className={'charts-auto-refresh-class'}>
+                    Auto Refresh <Switch defaultChecked={false} onChange={onChangeAutoRefresh} />
+                </div>
+                <div className={'charts-column-count-position-class'}>
+                    Column Count:{' '}
```
Comment (swaroopar) [inline src/components/content/monitor/MonitorChart.tsx:179]: did not understand the use of this field. We can remove it i think.

File: src/components/content/monitor/Monitor.tsx [hunk 2]
```diff
@@ -9,29 +9,30 @@ import { MonitorOutlined } from '@ant-design/icons';
...
 function Monitor(): JSX.Element {
     const [form] = Form.useForm();
     const [serviceId, setServiceId] = useState<string>('');
-    const [monitorChartItems, setMonitorChartItems] = useState<Tab[]>([]);
     const [deployedServiceList, setDeployedServiceList] = useState<ServiceVo[]>([]);
-    const [refreshMonitorMetrics, setRefreshMonitorMetrics] = useState<number | undefined>(undefined);
     const monitorMetricsQueue: Queue<Metric[]> = arrayQueue<Metric[]>(monitorMetricQueueSize);
+    const [refreshMonitorMetrics, setRefreshMonitorMetrics] = useState<number | undefined>(undefined);
+    const [monitorChartItems, setMonitorChartItems] = useState<Tab[]>([]);
+    const [isAutoRefresh, setIsAutoRefresh] = useState<boolean>(false);
```
[the remark this answers is not in the document]
Comment (Alice1319) [reply src/components/content/monitor/Monitor.tsx:26]: fixed.

File: src/components/content/monitor/Monitor.tsx [hunk 3]
```diff
@@ -72,66 +86,88 @@ function Monitor(): JSX.Element {
...
-                    cusServiceNameList.push(cusServiceName);
+                    customerServiceNameList.push(cusServiceName);
                 }
             });
-            setCustomerServiceNameList(cusServiceNameList);
         }
+        setCustomerServiceNameList(customerServiceNameList);
     };
 
     const handleChangeCustomerServiceName = (selectCustomerServiceName: string) => {
```
[the remark this answers is not in the document]
Comment (Alice1319) [reply src/components/content/monitor/Monitor.tsx:108]: fixed

File: src/components/content/monitor/Monitor.tsx [hunk 4]
```diff
@@ -72,66 +86,88 @@ function Monitor(): JSX.Element {
...
-            });
+    const onFinish = (values: { serviceName: string; customerServiceName: string; serviceId: string }) => {
+        if (!values.serviceId) {
+            return;
         }
-        return currentServiceId;
+
+        setServiceId(values.serviceId);
+    };
+    const autoRefreshHandler = (switchResult: boolean) => {
```
[the remark this answers is not in the document]
Comment (Alice1319) [reply src/components/content/monitor/Monitor.tsx:124]: fixed.

File: src/components/content/monitor/MonitorChart.tsx (hunk 1 above)
Comment (Alice1319) [reply src/components/content/monitor/MonitorChart.tsx:183]: Keep this one.

**Review by swaroopar — CHANGES_REQUESTED** [review]
column Count -> Name still not changed
Issues on small screen -
1. Radio buttons not clearly placed
2. Column Count field name and drop down not in same line.

**Review by swaroopar — CHANGES_REQUESTED** [review]
column Count -> Name still not changed
Issues on small screen -
1. Radio buttons not clearly placed
2. Column Count field name and drop down not in same line.

**Review by swaroopar — CHANGES_REQUESTED** [review]
still the metric duration is distorted on small screen. 
Also we can keep the metric type and resource name in one single line.

**Review by swaroopar — APPROVED** [review]
looks good.

**Pull request merged**